specification/azurefleet/AzureFleet.Management/baseVirtualMachineProfile.tsp

```diff
@@ -0,0 +1,2518 @@
+
+  /**
+   * Json formatted public settings for the extension.
+   */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Converted legacy object to typespec https://github.com/Azure/azure-rest-api-specs/blob/d715112415eb08572ee96cc4bd77217aa44dedf1/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachine.json#L2095C1-L2102C11"
+  settings?: Record<unknown>;
+
+  /**
+   * The extension can contain either protectedSettings or
+   * protectedSettingsFromKeyVault or no protected settings at all.
+   */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "Converted legacy object to typespec https://github.com/Azure/azure-rest-api-specs/blob/d715112415eb08572ee96cc4bd77217aa44dedf1/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachine.json#L2095C1-L2102C11"
+  protectedSettings?: Record<unknown>;
+
+  /**
+   * The provisioning state, which only appears in the response.
+   */
+  @visibility("read")
+  provisioningState?: string;
+
+  /**
+   * The virtual machine extension instance view.
+   */
+  instanceView?: VirtualMachineExtensionInstanceView;
+
+  /**
+   * Indicates whether failures stemming from the extension will be suppressed
+   * (Operational failures such as not connecting to the VM will not be suppressed
+   * regardless of this value). The default is false.
+   */
+  suppressFailures?: boolean;
+
+  /**
+   * The extensions protected settings that are passed by reference, and consumed
+   * from key vault
+   */
+  protectedSettingsFromKeyVault?: KeyVaultSecretReference;
+
+  /**
+   * Collection of extension names after which this extension needs to be
+   * provisioned.
+   */
+  provisionAfterExtensions?: string[];
+}
+
+/**
+ * The instance view of a virtual machine extension.
+ */
+model VirtualMachineExtensionInstanceView {
+  /**
+   * The virtual machine extension name.
+   */
+  name?: string;
+
+  /**
+   * Specifies the type of the extension; an example is "CustomScriptExtension".
+   */
+  type?: string;
+
+  /**
+   * Specifies the version of the script handler.
+   */
+  typeHandlerVersion?: string;
+
+  /**
+   * The resource status information.
+   */
+  @OpenAPI.extension("x-ms-identifiers", [])
+  substatuses?: InstanceViewStatus[];
+
+  /**
+   * The resource status information.
+   */
+  @OpenAPI.extension("x-ms-identifiers", [])
+  statuses?: InstanceViewStatus[];
+}
+
+/**
+ * Instance view status.
+ */
+model InstanceViewStatus {
+  /**
+   * The status code.
+   */
+  code?: string;
+
+  /**
+   * The level code.
+   */
+  level?: StatusLevelTypes;
+
+  /**
+   * The short localizable label for the status.
+   */
+  displayStatus?: string;
+
+  /**
+   * The detailed status message, including for alerts and error messages.
+   */
+  message?: string;
+
+  /**
+   * The time of the status.
+   */
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
```

Comment:
>FIXME [](http://example.com/codeflow?start=5&length=5)

I see a FIXME warning here.
Please verify you can remove it

Reply:
updated